### DISCUSSION TITLE
Limiting shemefully hoist to specific workspaces

### DISCUSSION THREAD

**Discussion by AGrzes** [pnpm/pnpm#3389] [Q&A]
I have monorepo where most projects (workspaces) can and should work with pnpm strict package structure but some use libraries that depend on flat structure and it is infeasible to modify them.

Now I must use `--shamefully-hoist`on top level install to get everything to work but side effect that none of the other projects benefit from strict dependencies .

It would be great to have option to hoist only for specific package dependencies and to hoist in package own node_modules instead of root one.

Are there any workarounds to the problem at the moment?

**Comment by emosheeep** [comment]
Similar to https://github.com/pnpm/pnpm/discussions/5439. You can use symlink-dir to hoist dependency you want manually.
